### PR TITLE
treewide: Add test cases for maximum chain-building depth

### LINKED ISSUE #58: Testcase model: allow setting maximum chain-building depth
Most X.509 validation APIs expose an API for setting a maximum-chain building depth, e.g. `X509_VERIFY_PARAM_set_depth` for OpenSSL. We should expose a similar setting in our testcase model and builder, to exercise these APIs.

NB: This depth normally refers to the maximum number of non-self-issued intermediate CA certificates that appear in the chain, i.e. is _generally_ 2 less than the actual maximal depth (which includes the end-entity and the trust anchor).

### PR DESCRIPTION
**Pull request by tetsuo-cpp**
Closes #58

- [x] Still need to update the OpenSSL and Go harnesses to use this new option.

### CODE REVIEW

File: limbo/testcases/rfc5280.py [hunk 1]
```diff
@@ -1613,3 +1613,130 @@ def malformed_subject_alternative_name(builder: Builder) -> None:
 
     builder = builder.server_validation()
     builder = builder.trusted_certs(root).peer_certificate(leaf).fails()
+
+
+@testcase
+def max_chain_depth_0(builder: Builder) -> None:
```
Comment (tetsuo-cpp) [inline limbo/testcases/rfc5280.py]: Actually, I just realised these don't belong in this file. Perhaps it'd make more sense to put them under `pathlen.py` since they're exercising similar functionality?

Comment (tetsuo-cpp) [reply limbo/testcases/rfc5280.py]: [c94f939](https://github.com/trailofbits/x509-limbo/pull/79/commits/c94f93992f5b4533a3168741c52b0c985907f891)

**Comment by tetsuo-cpp** [comment]
Hmm, I seem to be having some trouble regenerating `schema.go`:
```
tetsuo@Alexs-MacBook-Pro gocryptox509 % make schema.go
go generate
go: github.com/atombender/go-jsonschema/cmd/gojsonschema@latest: module github.com/atombender/go-jsonschema@latest found (v0.14.1), but does not contain package github.com/atombender/go-jsonschema/cmd/gojsonschema
main.go:3: running "go": exit status 1
make: *** [schema.go] Error 1
```
@woodruffw Does it work on your end? Might just be my setup as I haven't touched the Go harness before.

**Comment by woodruffw** [comment]
> @woodruffw Does it work on your end? Might just be my setup as I haven't touched the Go harness before.

Ah yeah, I also hit this -- I fixed it on #56, but that has other schema change issues so it can't be merged yet. 

Here's the relevant change: https://github.com/trailofbits/x509-limbo/pull/56/files#diff-b50ed33e7c6b9cca03fc536b22c16df8bc38bd9b9ed6e9ba07fc62a88c021e48L3-R3

(The upstream just changed the path that the codegen binary is at, everything else should be the same.)

**Review by woodruffw — APPROVED** [review]
LGTM, thank you!
